aptos-move/block-executor/src/scheduler.rs

```diff
@@ -49,6 +49,16 @@ impl ArmedLock {
     }
 }
 
+#[derive(Debug)]
+enum HaltedStatus {
```

Comment:
Strong opinion here:

The "halted" term has lost it's meaning long time ago, and is very, very, confusing. Why not rename it to just ExecutionStatus, with something like "Executing", "DoneProcessing", "Aborting". Or something similar.

Reply:
This PR is a bug fix. The thing was called halt.
ExecutionStatus also already exists.

Reply:
@gelash I'm standing by my opinion:

1. I just suggested some namings (ExecutionStage, ExecutionState are some other alternatives).
2. This PR introduces HaltedOk and HaltedErr (two very confusing terms). being a bug fix is not a reason not to give more relevant naming here.

Reply:
I think I can revert all this - since we don't care which error causes fallback - and maybe also just return empty error. Working on it right now.

Reply:
this is all gone back to the way you had it.